src/monorepo-shipit/src/Changeset.js

```diff
@@ -78,6 +83,19 @@ export default class Changeset {
     return this.__clone({ diffs });
   }
 
+  withDebugMessage(string: string, ...args: $ReadOnlyArray<string>): Changeset {
+    const messages = this.debugMessages ?? [];
+    messages.push(sprintf(string, ...args));
```

Comment:
This will push also into `this.debugMessages` since you are assigning the reference of this.debugMessages to const messages.

Not sure if that is what you want here. Did you mean to do `const messages = [...(this.debugMessages ?? [])]` instead?

Reply:
Changed using non-mutating `concat`, thanks! :)